influxdb/src/main/scala/akka/stream/alpakka/influxdb/impl/InfluxDBSourceStage.scala

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (C) 2016-2019 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.influxdb.impl
+
+import akka.annotation.InternalApi
+import akka.stream.alpakka.influxdb.InfluxDBSettings
+import akka.stream.{Attributes, Outlet, SourceShape}
+import akka.stream.stage.{GraphStage, GraphStageLogic, OutHandler}
+import org.influxdb.InfluxDB
+import org.influxdb.dto.{Query, QueryResult}
+import org.influxdb.impl.InfluxDBResultMapperHelper
+
+import scala.collection.JavaConverters._
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[influxdb] final class InfluxDBSourceStage[T](clazz: Class[T],
+                                                     settings: InfluxDBSettings,
+                                                     influxDB: InfluxDB,
+                                                     query: Query)
+    extends GraphStage[SourceShape[T]] {
+
+  val out: Outlet[T] = Outlet("InfluxDB.out")
+  override val shape = SourceShape(out)
+
+  override def createLogic(inheritedAttributes: Attributes): GraphStageLogic =
+    new InfluxDBSourceLogic[T](clazz, settings, influxDB, query, out, shape)
+
+}
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[influxdb] final class InfluxDBSourceLogic[T](clazz: Class[T],
+                                                     settings: InfluxDBSettings,
+                                                     influxDB: InfluxDB,
+                                                     query: Query,
+                                                     outlet: Outlet[T],
+                                                     shape: SourceShape[T])
+    extends GraphStageLogic(shape)
+    with OutHandler {
+
+  setHandler(outlet, this)
+
+  var dataRetrieved: Option[QueryResult] = None
+  var resultMapperHelper: InfluxDBResultMapperHelper = _
+
+  override def preStart(): Unit = {
+    resultMapperHelper = new InfluxDBResultMapperHelper
+    resultMapperHelper.cacheClassFields(clazz)
+
+    val queryResult = influxDB.query(query)
+    if (!queryResult.hasError) {
```

Comment:
What happens with the error?

Reply:
Due to the http api being capable of executing many queries there might be cases of a partially erroneous payload. 

For example

```bash
SELECT*FROM valid_measure; SELECT invalid_action FROM invalid_measure;
```

This will lead to a payload which contains data for the first query and error for the next.
My recent commit throws an error (failStage) when it comes to all queries from a request failing. In cases of partial errors the user shall be able to handle what went wrong and what went bad.
When it comes to the typed queries
which are bound to one type of data, thus one type of result, an error will be thrown anyway.